docs/aqua-book/language/modularity.md

```diff
@@ -0,0 +1,157 @@
+  result: i32  
+
+func fn() -> string:
+  <- CONST_NAME
+```
+
+To declare everything contained in the file, use `declares *`:
+
+```aqua
+-- This module declares `CONST_NAME`, `ServiceName`, `MyType` and `fn`
+aqua AquaFile declares *
+
+const CONST_NAME = "something"
+
+service ServiceName:
+  do_something()
+  
+data MyType:
+  result: i32  
+
+func fn() -> string:
+  <- CONST_NAME
+```
+
+Note that symbols declared with `declares` are not exported to the host language. To export symbols to the host language, use [`export`](#exporting-to-the-host-language-with-export).
+
+## Importing other modules
+
+Aqua modules can import other modules to use their [declarations](#specifying-what-is-declared-by-the-module-with-declares). There are two ways to import a module: with [`import`](#with-import) and [`use`](#with-use).
+
+### With `import`
+
+The main way to import a module is via `import`. Everything declared in the imported module comes into the current namespace directly.
+
+```aqua
+aqua AquaFile declares foo
+
+-- builtin.aqua declares `Op`
+import "@fluencelabs/aqua-lib/builtin.aqua"
+
+func foo():
+  Op.noop()
+```
+
+It is possible to cherry-pick and rename imports using `import ... from`:
+
+```aqua
+aqua AquaFile declares foo
+
+-- builtin.aqua declares `Op`
+import Op as Noop from "@fluencelabs/aqua-lib/builtin"
+
+func foo():
+  Noop.noop()
+```
+
+### With `use`
+
+The `use` expression makes it possible to import a module as a named scope. The name of the scope is taken from [`aqua` header](#giving-a-name-to-an-aqua-module-with-aqua) of the imported module. Everything declared in the imported module is available in the current namespace as a member of the scope.
+
+```aqua
+aqua AquaFile declares foo
+
+-- builtin.aqua's header is `aqua BuiltIn declares *`
+use "@fluencelabs/aqua-lib/builtin.aqua"
+
+func foo():
+  BuiltIn.Op.noop()
+```
+
+It is possible to rename the imported module with `use ... as ...` expression:
+
+```aqua
+use "@fluencelabs/aqua-lib/builtin" as Renamed
+
+func foo():
+  Renamed.Op.noop()
+```
+
+Creation of a scope with `use` makes it easier to avoid name clashes and to understand where the symbol comes from. Thus it is recommended to prefer `use` instead of `import` when possible.
+
+### Imports resolution
+
+To learn how compiler resolves the import path, see [JS Aqua API](./../aqua-js-api.md).
+
+:::info
+`.aqua` extension in `import` and `use` expressions can be omitted. So, `import "builtin.aqua"` does exactly the same as `import "builtin"`.
+:::
+
+## Exporting to the host language with `export`
+
+Inside Aqua language code modularity is achieved with [`declares`](#specifying-what-is-declared-by-the-module-with-declares), [`import`](#with-import) and [`use`](#with-use). However, what should be exported to the host language depends on the particular use case of aqua code and has nothing to do with code management inside Aqua. This is why exporting to the host language is a separate concept inside Aqua.
```

Comment:
I would refer to abilities as the mean to apply modularity on more fine grained level, as detalisation of aqua/declares/use.